src/models.py

```diff
@@ -88,38 +115,45 @@ class Dataset(BaseModel):
     status: StatusEnum
     user_id: str
     created_at: Optional[datetime] = None
-    
-    @field_serializer('created_at', mode='plain')
+
+    @field_serializer("created_at", mode="plain")
     def datetime_to_isoformat(field: datetime | None) -> str | None:
         if field is None:
             return None
         return field.isoformat()
-    
-    @field_validator('bbox', mode='before')
+
+    @field_validator("bbox", mode="before")
     @classmethod
     def transform_bbox(cls, raw_string: str | BoundingBox) -> BoundingBox:
         if isinstance(raw_string, str):
             # parse the string
-            s = raw_string.replace('BOX(', '').replace(')', '')
-            ll, ur = s.split(',')
-            left, bottom = ll.strip().split(' ')
-            right, upper = ur.strip().split(' ')
-            return BoundingBox(left=float(left), bottom=float(bottom), right=float(right), top=float(upper))
+            s = raw_string.replace("BOX(", "").replace(")", "")
+            ll, ur = s.split(",")
+            left, bottom = ll.strip().split(" ")
+            right, upper = ur.strip().split(" ")
```

Comment:
I think we should consistent with upper/top ... see line 147

Would stick to the way rasterio does it: https://rasterio.readthedocs.io/en/stable/api/rasterio.coords.html